src/lib/pci_virtio_sock.c

```diff
@@ -1644,6 +1672,141 @@ static void pci_vtsock_notify_tx(void *vsc, struct vqueue_info *vq)
 	kick_tx(sc, "notify");
 }
 
+static void apply_socket_config(__unused struct pci_vtsock_sock *s, void *buf)
+{
+	struct vtsock_config_hdr *hdr = (struct vtsock_config_hdr *)buf;
+
+	if (hdr->version != 1) {
+		fprintf(stderr, "config: ERROR "
+			"could not apply unknown config version %d!\n",
+			hdr->version);
+		return;
+	}
+
+}
+
+static void handle_socket_config(struct pci_vtsock_sock *s, int fd)
+{
+	ssize_t len;
+	uint32_t sz;
+	char *szp = (char *)&sz;
+	uint32_t read_upto;
+	void *buf;
+
+ next_message:
+	read_upto = 0;
+
+	do {
+		len = read(fd, szp + read_upto, sizeof(uint32_t) - read_upto);
+		if (len == -1) {
+			if (errno == EAGAIN)
+				continue;
+			fprintf(stderr, "config: ERROR "
+				"%s while reading config message length\n",
+				strerror(errno));
+			return;
+		}
+		if (len == 0) {
+			if (read_upto)
+				fprintf(stderr, "config: ERROR "
+					"EOF during "
+					"config message length read\n");
+			return;
+		}
+		read_upto += len;
+	} while (read_upto < sizeof(uint32_t));
+
+	if (sz < sizeof(struct vtsock_config_hdr)) {
+		fprintf(stderr, "config: ERROR "
+			"message length %d is less than header size %lu\n",
+			sz, sizeof(struct vtsock_config_hdr));
+		return;
+	}
+
+	buf = malloc(sz);
+	if (buf == NULL) {
+		fprintf(stderr, "config: ERROR "
+			"%s from config message buffer malloc\n",
+			strerror(errno));
+		return;
+	}
+	((struct vtsock_config_hdr *)buf)->len = sz;
+
+	do {
+		len = read(fd, ((char *)buf) + read_upto, sz - read_upto);
+		if (len == -1) {
+			if (errno == EAGAIN)
+				continue;
+			fprintf(stderr, "config: ERROR "
+				"%s while reading config message\n",
+				strerror(errno));
+			goto err;
+		}
+		if (len == 0) {
+			fprintf(stderr, "config: ERROR "
+				"EOF during config message body read\n");
+			goto err;
+		}
+		read_upto += len;
+	} while (read_upto < sz);
+
+	apply_socket_config(s, buf);
+
+	free(buf);
+	goto next_message;
+
+ err:
+	free(buf);
+	return;
+}
+
+/*
+ * Check for an attached socket configuration fd, s->mtx must be held
+ *
+ * If a configuration fd is attached, this will block until the
+ * configuration request is handled!
+ *
+ * Returns > 0 for regular messages, 0 for EOF, -1 for error, and -2
+ * for configuration messages.
+ */
+static ssize_t readrx(struct pci_vtsock_sock *s,
+		      struct iovec *iov, int iov_len)
+{
+	ssize_t len;
+	char control[CMSG_SPACE(sizeof(int /* fd */))];
+	struct msghdr msghdr = {
+		.msg_name = NULL,
+		.msg_namelen = 0,
+		.msg_iov = iov,
+		.msg_iovlen = iov_len,
+		.msg_control = control,
+		.msg_controllen = CMSG_SPACE(sizeof(int /* fd */)),
+	};
+	struct cmsghdr *cmsg;
+	int fd;
+
+	len = recvmsg(s->fd, &msghdr, 0);
+	if (s->configurable && len > 0 && msghdr.msg_controllen > 0) {
```

Comment:
It could be an error or it could continue to be ignored as it is currently. I'm happy to make it an error as it's a strong sign that You're Doing It Wrong.